src/main/java/net/openhft/chronicle/bytes/RandomDataOutput.java

```diff
@@ -31,7 +31,7 @@
 import static net.openhft.chronicle.core.util.ObjectUtils.requireNonNull;
 
 @SuppressWarnings({"rawtypes", "unchecked"})
-public interface RandomDataOutput<R extends RandomDataOutput<R>> extends RandomCommon {
+public interface RandomDataOutput<R extends RandomDataOutput<R>> extends RandomDataInput, RandomCommon {
```

Comment:
sure you want to extend RandomDataInput?

Reply:
Yes, this is somewhat conceptual change if you have access to write you should have access to read as well, for example in order to do CAS you need both... but that's not the reason I do it this way.

I would compare this to List vs Immutable versions of it, in java this sucks because both are shared around with LIst interface, but for example, in kotlin we have List (which has read-only methods) and MutableList which has read+write (equivalent to java List)... 

It would not make much sense to create a List that has only write methods, you'd have to blindly write to it, what if its a linked list and you want to peek... 

It's the same concept here. If you have to write access to buffer you should have read as well, if you have read access you should not be able to write.

Reply:
There could be an underlying source (like Wire) and then we could derive two different views of the underlying source: one for reading and one for writing. The latter can only be used for writing but will mutate the underlying source. These mutations can be picked up bu the reading view and directly from the underlying source.

Reply:
Had a discussion with Per, 
and will post a separate PR that addresses this change with the suggestions that he made
I think it will boil down to what is conceptually preferable?
this PR = Reader can only read, the writer can both read and write
the other PR = Reader only reads, Writers can only write, if you want both (do a typecast)

Reply:
> Had a discussion with Per, and will post a separate PR that addresses this change with the suggestions that he made I think it will boil down to what is conceptually preferable? this PR = Reader can only read, the writer can both read and write the other PR = Reader only reads, Writers can only write, if you want both (do a typecast)

do a typecast or use an entity that implements both

Reply:
This is the alternative that does not change the meaning of the writer interfaces, so writers stay write only but whenever we have a writer and want to read we have to do a typecast https://github.com/OpenHFT/Chronicle-Bytes/pull/362 (and this is how it would look like when I actually remove the deprecations https://github.com/OpenHFT/Chronicle-Bytes/commit/8d224d85c7f0e660f1961debe56741f605d3b993).

Reply:
I think my vote is for https://github.com/OpenHFT/Chronicle-Bytes/commit/8d224d85c7f0e660f1961debe56741f605d3b993

Reply:
ok then please @JerryShea @minborg look at this PR - https://github.com/OpenHFT/Chronicle-Bytes/pull/362
i'll close this one later